tests/test_loader.py

```diff
@@ -289,16 +325,20 @@ async def test_get_config_flows(hass):
 async def test_get_zeroconf(hass):
     """Verify that custom components with zeroconf are found."""
     test_1_integration = _get_test_integration(hass, "test_1", True)
-    test_2_integration = _get_test_integration(hass, "test_2", True)
+    test_2_integration = _get_test_integration_with_zeroconf_matcher(
+        hass, "test_2", True
+    )
 
     with patch("homeassistant.loader.async_get_custom_components") as mock_get:
         mock_get.return_value = {
             "test_1": test_1_integration,
             "test_2": test_2_integration,
         }
         zeroconf = await loader.async_get_zeroconf(hass)
-        assert zeroconf["_test_1._tcp.local."] == ["test_1"]
-        assert zeroconf["_test_2._tcp.local."] == ["test_2"]
+        assert zeroconf["_test_1._tcp.local."] == [{"domain": "test_1"}]
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6158285)